src/sagemaker/predictor.py

```diff
@@ -406,6 +413,88 @@ def _get_model_names(self):
         return [d["ModelName"] for d in production_variants]
 
 
+class LegacySerializer(BaseSerializer):
+    """Wrapper that makes legacy serializers forward compatibile."""
+
+    def __init__(self, serializer):
+        """Placeholder docstring.
+
+        Args:
+            serializer (callable): A legacy serializer.
+        """
+        self.serializer = serializer
+        self.content_type = getattr(serializer, "content_type", None)
+
+    def __call__(self, *args, **kwargs):
```

Comment:
are the function signatures that different across legacy implementations? I think it would be better to avoid `*args` and `**kwargs` here if possible.

Reply:
Some of the classes have different function signatures. For example, `_NPYSerializer` has the signature `def __call__(self, data, dtype=None)`. If I were to not use *args and **kwargs, then I would need to make changes (or remove) some tests. Since LegacySerializer and LegacyDeserializer are temporary fixes so that I can make incremental changes, IMO I think it's fine.